src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Max.cs

```diff
@@ -273,13 +274,104 @@ private static Vector512<T> IsNegative<T>(Vector512<T> vector)
             }
 
             return Vector512.LessThan(vector, Vector512<T>.Zero);
+#endif
         }
 
-        /// <remarks>
-        /// This is the same as <see cref="Aggregate{T, TTransformOperator, TAggregationOperator}(ReadOnlySpan{T})"/>
-        /// with an identity transform, except it early exits on NaN.
-        /// </remarks>
-        private static T MinMaxCore<T, TMinMaxOperator>(ReadOnlySpan<T> x)
+        /// <summary>Gets whether each specified <see cref="float"/> is positive.</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector128<T> IsPositive<T>(Vector128<T> vector)
+        {
+#if NET9_0_OR_GREATER
+            return Vector128.IsPositive(vector);
+#else
+            if ((typeof(T) == typeof(byte))
+             || (typeof(T) == typeof(ushort))
+             || (typeof(T) == typeof(uint))
+             || (typeof(T) == typeof(ulong))
+             || (typeof(T) == typeof(nuint)))
+            {
+                return Vector128<T>.AllBitsSet;
+            }
+
+            if (typeof(T) == typeof(float))
+            {
+                return Vector128.GreaterThanOrEqual(vector.AsInt32(), Vector128<int>.Zero).As<int, T>();
+            }
+
+            if (typeof(T) == typeof(double))
+            {
+                return Vector128.GreaterThanOrEqual(vector.AsInt64(), Vector128<long>.Zero).As<long, T>();
+            }
+
+            return Vector128.GreaterThanOrEqual(vector, Vector128<T>.Zero);
+#endif
+        }
+
+        /// <summary>Gets whether each specified <see cref="float"/> is positive.</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector256<T> IsPositive<T>(Vector256<T> vector)
+        {
+#if NET9_0_OR_GREATER
+            return Vector256.IsPositive(vector);
+#else
+            if ((typeof(T) == typeof(byte))
+             || (typeof(T) == typeof(ushort))
+             || (typeof(T) == typeof(uint))
+             || (typeof(T) == typeof(ulong))
+             || (typeof(T) == typeof(nuint)))
+            {
+                return Vector256<T>.AllBitsSet;
+            }
+
+            if (typeof(T) == typeof(float))
+            {
+                return Vector256.GreaterThanOrEqual(vector.AsInt32(), Vector256<int>.Zero).As<int, T>();
+            }
+
+            if (typeof(T) == typeof(double))
+            {
+                return Vector256.GreaterThanOrEqual(vector.AsInt64(), Vector256<long>.Zero).As<long, T>();
+            }
+
+            return Vector256.GreaterThanOrEqual(vector, Vector256<T>.Zero);
+#endif
+        }
+
+        /// <summary>Gets whether each specified <see cref="float"/> is positive.</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static Vector512<T> IsPositive<T>(Vector512<T> vector)
+        {
+#if NET9_0_OR_GREATER
+            return Vector512.IsPositive(vector);
```

Comment:
Do we want public TensorPrimitives APIs for IsPositive/IsNegative?

Reply:
Those are part of P3 of for the Tensor API review, we didn't finish getting through them in the one we did last Thursday, so not exposing them publicly yet.